tests~/MultiDictionaryTests.cs

```diff
@@ -0,0 +1,239 @@
+using System.Diagnostics;
+using CsCheck;
+using SpacetimeDB;
+using Xunit;
+
+public class MultiDictionaryTests
+{
+    /// <summary>
+    /// Generate a list of KeyValuePairs.
+    /// If any two items of the list have duplicate Keys, they are guaranteed to have duplicate Values.
+    /// </summary>
+    /// <typeparam name="TKey"></typeparam>
+    /// <typeparam name="TValue"></typeparam>
+    /// <param name="g1"></param>
+    /// <param name="g2"></param>
+    /// <param name="equality"></param>
+    /// <returns></returns>
+    Gen<List<KeyValuePair<TKey, TValue>>> ListWithNormalizedDuplicates<TKey, TValue>(Gen<TKey> g1, Gen<TValue> g2, IEqualityComparer<TKey> equality, int ListMinLength = 0, int ListMaxLength = 32)
```

Comment:
The only thing that really matters about this is that it returns `Gen<List<KeyValuePair<TKey, TValue>>>` which is an object that randomly generates `List<KeyValuePair<TKey, TValue>>`s.